### ISSUE TITLE
Remove `toChannelGroupDto` conversion from status-desktop

### ISSUE THREAD

**Issue by osmaczko** [status-im/status-desktop#12595]
### Description

The [toChannelGroupDto](https://github.com/status-im/status-desktop/blob/b414d616a672238fee9ffc2afd3a15346bf3656a/src/app_service/service/community/dto/community.nim#L525-L553) function in the desktop is essentially a redundant replication of [getChannelGroups](https://github.com/status-im/status-go/blob/eb437e9d8dd5bd1ca3b18fa05c4853d6ae8f4956/services/chat/api.go#L214-L237) from status-go. This isn't just a mere duplication of logic; it leads to numerous data inconsistency issues. Sometimes, we propagate `ChannelGroupDto` directly from status-go, and at other times, we run our conversion. This inconsistency opens up potential mismatches between the two. Ideally, we should maintain synchronization with the status-go version. However, given that status-go is platform-agnostic and can be altered by any team (such as mobile), ensuring consistent synchronization is impractical. Any deviation can be incredibly challenging to debug. When issues arise, tracing them back to the root cause, particularly if one lacks experience, is like searching for a needle in a haystack, with `toChannelGroupDto` being the last suspected culprit.

The "ChannelGroups" concept in status-go was intended to unify the sections. But, `ChannelGroup` isn't really used much by clients. It's used at first when getting sections, but after that, specific types like `Community` are used, which means the desktop has to change how it handles things.

There are a few ways to solve this:

A) Remove `toChannelGroupDto` from the desktop and rely on status-go's conversion (this would require creating a new API, such as `communityToChannelGroup(communityID)`).
B) Do away with the "ChannelGroups" idea in status-go and retain it solely within status-desktop.
C) Modify all status-go signals to be of type `ChannelGroup`, either as a replacement for or in addition to `Community`.

**Comment by osmaczko** [comment]
CC: @jrainville

**Comment by jrainville** [comment]
Yeah, you hit it on the head. The concept of channel groups was to unify the personal chats and communities under a common type, but we end up only using it on app start when we load the sections, because the personal section is basically only loaded once and then no longer needed.

In theory,  the third (C) option is the best in the long run, since we'd only deal with Channel Groups. However, there are multiple possible problems with that.
First, it's a lot of work, since we return the Community type a lot.
Second, mobile also uses all those Community returning APIs, so we'd have to deprecate a lot of APIs and force them to update to ChannelGroup.
Thirdly, ChannelGroup returns a lot more data than Community, so all the APIs that used to return Community would need to pass through a conversion function that would slow down the app  a little bit.

So I think the third option is not worth it, even though it could help get everything standardized.

I think we should probably do away with the concept of ChannelGroups all together. It was a nice idea in theory to unify the sections, but in the end, it just creates more confusion and complexity.

We can go back to what we had before and/or see what mobile does to fetch the sections at the start and do the same.

It might just be about getting all communities, and getting the personal chat section in parallel separately
